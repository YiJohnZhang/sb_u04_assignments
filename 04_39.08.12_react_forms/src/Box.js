import React from 'react';

function Box({width, height, backgroundColor, deleteThisBox}){

	const boxStyle = {width, height, backgroundColor}

	return(

		<div style={boxStyle}>

			<button onClick={deleteThisBox}>x</button>

		</div>

	);

}

export default Box;