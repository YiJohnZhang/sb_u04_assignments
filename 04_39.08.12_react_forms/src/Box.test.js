import React from 'react';
import {render, fireEvent} from '@testing-library/react';
import Box from './Box';

test('smoke test', () => {

	render(<Box />);

});

test('snapshot', () => {
	
	const {asFragment} = render(<Box />);
	expect(asFragment()).toMatchSnapshot();

});