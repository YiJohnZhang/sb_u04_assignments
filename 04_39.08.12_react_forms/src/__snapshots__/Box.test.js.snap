// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<DocumentFragment>
  <div>
    <button>
      x
    </button>
  </div>
</DocumentFragment>
`;
